pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -184,6 +185,21 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, r.addClusterDeploymentFinalizer(cd)
 	}
 
+	// Ensure we have an AMI set, if not lookup the latest:
```

Comment:
Just a thought ... when we add support for other cloud providers, it may make sense to remove this code from here and simply call a function specific to the cloud provider. Otherwise, we'll likely end up cluttering this reconcile loop.

Reply:
Sounds good to me.